### ISSUE TITLE
configure cannot handle GCC correctly

### ISSUE THREAD

**Issue by barracuda156** [yxlin/ggdmc#6]
Configure script does not work correctly: it fails to identify `gcc` as `gcc` and cannot understand that `gcc` supports OpenMP.
```
* installing *source* package ‘ggdmc’ ...
** using staged installation
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether /opt/local/bin/g++-mp-12 -std=gnu++17 accepts -g... yes
checking how to run the C++ preprocessor... /opt/local/bin/g++-mp-12 -std=gnu++17 -E
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether /opt/local/bin/g++-mp-12 -std=gnu++17 accepts -g... (cached) yes
checking whether g++ version is sufficient... almost
configure: WARNING: Compiler self-identifies as being compliant with GNUC extensions but is not g++.
checking for macOS... found
checking for macOS Apple compiler... not found
checking for clang compiler... not found
configure: WARNING: unsupported macOS build detected; if anything breaks, you keep the pieces.
checking LAPACK_LIBS... R-supplied partial LAPACK found
configure: WARNING: Some complex-valued LAPACK functions may not be available
configure: creating ./config.status
config.status: creating inst/include/RcppArmadilloConfigGenerated.h
config.status: creating src/Makevars
** libs
using C++ compiler: ‘g++-mp-12 (MacPorts gcc12 12.3.0_0) 12.3.0’
using C++11
```

**Comment by barracuda156** [comment]
OK, the problem is that it thinks the latest GCC is 10.x. Once that is fixed, GCC is recognized, however OpenMP is still broken:
```
* installing *source* package ‘ggdmc’ ...
** using staged installation
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether /opt/local/bin/g++-mp-12 -std=gnu++17 accepts -g... yes
checking how to run the C++ preprocessor... /opt/local/bin/g++-mp-12 -std=gnu++17 -E
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether /opt/local/bin/g++-mp-12 -std=gnu++17 accepts -g... (cached) yes
checking whether g++ version is sufficient... yes, with OpenMP as version 12.3.0
checking for macOS... found
checking for macOS Apple compiler... not found
checking for clang compiler... not found
configure: WARNING: unsupported macOS build detected; if anything breaks, you keep the pieces.
checking LAPACK_LIBS... R-supplied partial LAPACK found
configure: WARNING: Some complex-valued LAPACK functions may not be available
checking for OpenMP... missing
configure: creating ./config.status
config.status: creating inst/include/RcppArmadilloConfigGenerated.h
config.status: creating src/Makevars
** libs
using C++ compiler: ‘g++-mp-12 (MacPorts gcc12 12.3.0_0) 12.3.0’
using C++11
```

**Comment by yxlin** [comment]
Hi there, I am sorry it does not work on your case. 

From your output, it should be a problem due to some changes in the macOS setting. Previously, I modified Dirk Eddelbuettel's `configure.ac` in the RccArmadillo to manually turn off the Open MP when it detected a macOS system. I borrowed a Mac machine to test that, but I do not have the resources to do that at the moment. I suggest you modify the `configure.ac` if you must install the software on a Mac machine. A new `configure.ac` should be able to adapt to the new macOS change. 

HTH.
